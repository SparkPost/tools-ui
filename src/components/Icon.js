import React from 'react';
import classNames from 'classnames';

const iconPrefix = 'fa-';

const mapPropsToClasses = ({ name, size, extras = [] }) => (
  classNames('fa', {
    [`${iconPrefix}${name}`]: name,
    [`${iconPrefix}${size}`]: size
  }, extras)
);

// eslint-disable-next-line import/no-anonymous-default-export
export default (props) => <i className={mapPropsToClasses(props)} />;
