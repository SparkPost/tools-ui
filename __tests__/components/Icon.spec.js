import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../../src/components/Icon';

describe('Icon', () => {

  it('should render correctly with all options', () => {
    expect(renderer.create(<Icon
      name='awesome-icon'
      size='lg'
      extras={['spin', 'big', 'super-special']}
    />)).toMatchSnapshot();
  });

  it('should render correctly with no options', () => {
    const someFunction = () => {};
    expect(renderer.create(<Icon />)).toMatchSnapshot();
  });

});
