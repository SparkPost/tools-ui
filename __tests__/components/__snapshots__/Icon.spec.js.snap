exports[`Icon should render correctly with all options 1`] = `
<i
  className="fa fa-awesome-icon fa-lg fa-spin fa-big fa-super-special" />
`;

exports[`Icon should render correctly with no options 1`] = `
<i
  className="fa " />
`;
